app/Http/Controllers/GoodController.php

```diff
@@ -129,9 +140,11 @@ public function deleteGood(Request $request, $good_id)
        * @return Response
        * @description Delete a specify good.
        */
-        if(!$request->session()->has('userid')) return Redirect::back();
+		if(!$request->session()->has('userid')) 
+			 return Redirect::back();
         $good = GoodInfo::find($good_id);
-        if($request->session()->get('userid')!=$good->user_id) return Redirect::back();
+		if($request->session()->get('userid')!=$good->user_id && !$request->session()->has('admin')) 
+			 return Redirect::back();
         $good->delete();
         return redirect('/good');
```

Comment:
Please use Redirect::to() instead of redirect() helper function
make all the redirect to url logic use the same function